e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
+            # Update the handler's "table" if the frequency is not monthly.
+            if not simple and freq != "mon":
+                available_handlers[key]["table"] = _get_table_for_non_monthly_freq(
+                    handler["name"], handler["table"], freq, realm, tables_path
+                )
+
+        print_message(
+            f"Loaded all available handlers with a frequency of {freq}.",
+            "debug",
+        )
+        chosen_handlers = [handler for handler in available_handlers.values()]
+
+        return chosen_handlers
+
+    for var in var_list:
+        # Use the high frequency version of a variable handler based on the
+        # specified frequency. For example, for "pr" with a frequency of "day"
+        # or "3hr, use "pr_highfreq.py" instead of "pr.py".
+        # TODO: Once the handler modules are refactored using classes, we can
+        # update this check to something more sophisticated.
+        if _use_highfreq_handler(var, freq):
+            var += "_highfreq"
+
+        handler = available_handlers.get(var)
+        if handler is None:
+            raise KeyError(
+                f"Variable '{var}' failed to load. Either an entry is not defined in "
+                "the default handlers file, or a Python module of the same name does "
+                "not exist."
+            )
+
+        # Update the handler's "table" if the frequency is not monthly.
+        if not simple and freq != "mon":
+            handler["table"] = _get_table_for_non_monthly_freq(
+                handler["name"], handler["table"], freq, realm, tables_path
+            )
+
+        chosen_handlers.append(handler)
+        print_message(
+            f"Loaded `{var}` handler for variable `{handler['name']}`.", "debug"
+        )
+
+    return chosen_handlers
+
+
+def _get_available_handlers(handlers_path: str) -> Dict[str, Dict[str, str]]:
+    """Loads default and/or complex handlers based on a list of variable names.
 
-    Returns a Path to the correct table, and if the variable is included in the selected table
+    Parameters
+    ----------
+    handlers_path : str
+        The path to the complex handlers, which are Python modules.
+
+    Returns
+    -------
+    Dict[str, Dict[str, str]]
+        A dictionary of dictionaries, with each dictionary storing the
+        information for a complex or default handler.
     """
-    real_table = get_table_freq(table, freq)
-    if not real_table:
-        return None, None
-    table_path = Path(tables, real_table)
-    if not table_path.exists():
-        return None, None
-    table_data = get_table_info(tables, real_table)
+    default_handlers = _get_default_handlers()
+    complex_handlers = _get_complex_handlers(handlers_path)
+    all_handlers = {**default_handlers, **complex_handlers}
 
-    if variable not in table_data['variable_entry'].keys():
-        return table_path.name, False
-    else:
-        return table_path.name, True
+    return all_handlers
```

Comment:
This function combines the default handlers and complex handlers as a single dictionary.